volume_project/extract/2022/miovision_sqls/miovision_centreline_20220705.sql

```diff
@@ -0,0 +1,173 @@
+-- this is a bit of a convoluded process... here are the steps I took. I'm happy to make it more efficient
+-- 1. wrote code to find the centreline_ids that correspond to the segments that touch miovision intersections; put these in a table
+-- 2. did some error checking to see if that code worked (there were 9 situations where it didn't)...
+-- 3. manually investigated and fixed the centrelines that needed to be fixed (in the table)
+
+-- 1. Here is the code that finds centreline_ids that touch miovision intersections:
+CREATE TABLE scannon.centreline_miovision_20220705 AS (
```

Comment:
```
select  intersection_uid, leg, count(1) 
from  scannon.centreline_miovision_20220705
group by intersection_uid, leg
having count(1) >1
```
returned four intersections + leg combination that got tagged to more than 1 centreline_id
92	"E"	2
71	"E"	2
91	"W"	2
97	"S"	2

Reply:
I checked these out. 

For 3 of them (92, 71 and 97) I updated the direction to match the (correct) `centreline_id` (in other words, where two legs had been given an 'E', one should've been an 'S', so I updated the table accordingly.

For 91 (Lakeshore and Spadina), there are actually 5 centreline segments and two are west legs - one is Lakeshore and the other is the ramp to the Gardiner. How is this handled in other situations? This is a relatively new intersection.

Reply:
...and since it's new it wasn't around in 2022 so I ran the miovision mat view with the updated `scannon.miovision_centreline_20220705` table.

Reply:
if its lakeshore and spadina then i guess lakeshore ? 🤔

Reply:
This still won't affect TEPS 2023 because the intersection was only up and running in 2024, but yeah... weird intersection. Both of those western legs are measured by the camera.
![image](https://github.com/CityofToronto/bdit_volumes/assets/80077912/7f50fa58-a093-46a9-8934-62a59fe91be5)

Reply:
Wut...
We need a way to distinguish between those two legs (north and south western legs?)